### ISSUE TITLE
Scale Item while Dragging

### ISSUE THREAD

**Issue by lkevin029** [h6ah4i/android-advancedrecyclerview#193]
How to make the item bigger when dragging start?
this is my code. 

            mRecyclerViewDragDropManager.setOnItemDragEventListener(new RecyclerViewDragDropManager.OnItemDragEventListener() {
            @Override
            public void onItemDragStarted(int position) {
                View v = mRecyclerView.getChildAt(position);
                ScaleAnimation animation = new ScaleAnimation(1, 1.2f, 1, 1.2f, Animation.RELATIVE_TO_SELF, 1, Animation.RELATIVE_TO_SELF, 1);
                v.setAlpha(0.5f);
                animation.setFillAfter(true);
                v.startAnimation(animation);

            }

            @Override
            public void onItemDragPositionChanged(int fromPosition, int toPosition) {

            }

            @Override
            public void onItemDragFinished(int fromPosition, int toPosition, boolean result) {
            }
        });

but its not working because the holder in dragging is different its just a copy of the original one.

I think its better you add holder in onItemDrag events so every one can make any changes in Item while dragging

**Comment by h6ah4i** [comment]
Hi. Thanks for your suggestion, but that is difficult to implement `onItemDrag()` like events.

1. views and its holder objects will be recycled, so should not implement callback to holders
2. in generally, item views should be manipulated in `onBindViewHolder()` method only
3. dragging item is rendered as `RecyclerView.ItemDecoration`, so `ObjectAnimator` won't work

However, there is a way to implement scaling animation of the dragging item. You can achieve that by modifying the `DraggingItemDecorator` class of this library.

https://github.com/h6ah4i/android-advancedrecyclerview/blob/master/library/src/main/java/com/h6ah4i/android/widget/advrecyclerview/draggable/DraggingItemDecorator.java

**Comment by lkevin029** [comment]
thank you for your reply this was helpful..
I got one more question 
I got a gridlayout recyclerview 
numbers  = span

1  1   2   1 1  

I can't drag the 2 in the corner its on purpose or it is a bug ?
lets just say 2 span width is bigger than 1 so i can't drag it in the 1st element of the row and last element...
sorry for my english >,<

**Comment by h6ah4i** [comment]
@lkevin029 Thanks for pointing it out. I have just fixed the behavior for GridLayoutManager. Try the patched version of the library via jitpack.


#### build.gradle
```gradle
repositories {
    maven { url "https://jitpack.io" }
}

dependencies {
    compile 'com.github.h6ah4i:android-advancedrecyclerview:fbb4f3bc883de883d79ec04d03d9cf74f2773322'
}
```
branch: https://github.com/h6ah4i/android-advancedrecyclerview/tree/feature/issue_193_improve_grid_layout_drag

**Comment by h6ah4i** [comment]
@lkevin029 The `feature/issue_193_improve_grid_layout_drag` branch has been merged into master branch now and released as v.0.8.7. Thanks.

**Comment by richardjbs** [comment]
@h6ah4i I am trying to implement your suggestion to be able to scale the dragged view while dragging. I was trying to customize the DraggingItemDecorator. The issue is that the decorator is being use by the RecyclerViewDragDropManager internally on the startDragging() which is a private method. So extending from RecyclerViewDragDropManager to override startDragging and specify a custom DraggingItemDecorator is not an option. A lot of classes within the library are protected so I cannot use them on my project. I don't want to include all the source code on my project. I was trying to keep using gradle to import it dynamically. Any suggestion on this? How should I proceed to specify the DraggingItemDecorator implementation that I want to use to the RecyclerViewDragDropManager?

Thanks

**Comment by h6ah4i** [comment]
@richardjbs Hi. Yes, I meant that my suggestion require to fork this project. I admit that forking & modifying `DraggingItemDecorator` is not the best option. I think creating a new stable classes or API is needed, because `DraggingItemDecorator` is tightly integrated to `RecyclerViewDragDropManager` and it will be changed in future.

Also, I will reopen this issue. I closed this issue by mistake, I have fixed the sub-issue in v0.8.7 but the main issue is still not fixed yet.

Thanks

**Comment by h6ah4i** [comment]
Related #292.

**Comment by h6ah4i** [comment]
This enhancement has been included in v0.10.0.

**Comment by egek92** [comment]
setDraggingItemScale(float scale) works perfect but it only draws inside the recyclerView bounds, is it possible to make it get bigger on outside as well?
